on-chain-nft/io/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+#![no_std]
+
+use gear_lib::non_fungible_token::{royalties::*, state::*, token::*};
+use gstd::{prelude::*, ActorId};
+
+pub type LayerId = u128;
+pub type ItemId = u128;
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum OnChainNFTQuery {
+    /// Returns an NFT for a specified `token_id`.
+    /// Requirements:
+    /// * `token_id` MUST exist
+    /// Arguments:
+    /// * `token_id` - is the id of the NFT
+    /// On success, returns TokenURI struct.
+    TokenURI { token_id: TokenId },
+    /// Base NFT query. Derived from gear-lib.
+    Base(NFTQuery),
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum OnChainNFTAction {
+    /// Mints an NFT.
+    /// Requirements:
+    /// * `description` MUST contain layers and layers' items that EXIST
+    /// Arguments:
+    /// * `token_metadata` - is a default token metadata from gear-lib.
+    /// * `description` - is the vector of layer's item id, where
+    /// the index i is the layer id.
+    /// On success, returns NFTEvent::Mint from gear-lib.
```

Comment:
More details: mints NFT consisted from layers..
```suggestion
    /// Mints an NFT.
    ///
    /// Requirements:
    /// * `description` MUST contain layers and layers' items that EXIST
    ///
    /// Arguments:
    /// * `token_metadata` - is a default token metadata from gear-lib.
    /// * `description` - is the vector of layer's item id, where
    /// the index i is the layer id.
    ///
    /// On success, returns NFTEvent::Mint from gear-lib.
```